File: src/test/programmers/level1/HashadSu.java
package test.programmers.level1;

public class HashadSu {
    public static void main(String[] args) {
        System.out.println(solution(11));
    }

    static boolean solution(int x) {
        boolean answer = true;
        int sum = 0;
        char[] chars = String.valueOf(x).toCharArray();

        for(char ch : chars) {
            sum += ch - '0';
        }

        return answer = x % sum == 0 ? true : false;
    }
}

File: src/test/programmers/python/level1/HashadSu.py
def solution(x):
    x = str(x)
    a = 0
    for i in range(len(x)):
        a += int(x[i])
    return True if int(x) % a == 0 else False

# 다른 사람 풀이
# def solution(x):
#     return x % sum([int(i) for i in str(x)]) == 0

